scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/utils/TaskKiller.java

```diff
@@ -67,16 +66,35 @@ private void interruptRepeatedly() {
         }
     }
 
+    private long getCleanupTimeMilliseconds() {
+        try {
+            String cleanupTimeString = System.getProperty("proactive.task.cleanup.time");
+            if (cleanupTimeString != null) {
+                return Long.parseLong(cleanupTimeString);
+            } else {
+                return CLEANUP_TIME_DEFAULT_MILLISECONDS;
+            }
+        } catch (NumberFormatException e) {
+            LOGGER.warn("proactive.task.cleanup.time: "
+                    + System.getProperty("proactive.task.cleanup.time")
+                    + " is not parseable to long: "
+                    + e.getMessage());
+            return CLEANUP_TIME_DEFAULT_MILLISECONDS;
+        }
+    }
+
+    private Sleeper getCleanupTimeSleeper() {
+        long cleanupTimeMilliseconds = this.getCleanupTimeMilliseconds();
+        return new Sleeper(cleanupTimeMilliseconds, LOGGER);
+    }
+
     /**
      * Interrupts and gives the thread time to cleanup
      */
     private void interruptAndWaitCleanupTime() {
         threadToKill.interrupt();
-        int iterationsToHitWaitingTimeLimit =
-                SECONDS_TO_WAIT_UNTIL_AGGRESSIVELY_INTERRUPT_TASK * THREAD_INTERRUPT_SLEEP_TO_SECONDS_MULTIPLIER;
-        for (int i = 0; threadToKill.isAlive() && i < iterationsToHitWaitingTimeLimit; i++) {
-            threadInterruptSleeper.sleep();
-        }
+        Sleeper cleanupTimeSleeper = getCleanupTimeSleeper();
+        cleanupTimeSleeper.sleep();
```

Comment:
The problem with this implementation, is that the sleeper will wait for 10 or more seconds no matter if the threads actually finish quickly.